lib/streamlit/elements/widgets/number_input.py

```diff
@@ -51,6 +51,8 @@
 
 
 Number: TypeAlias = Union[int, float]
+_IntOrNone = TypeVar("_IntOrNone", int, None)
```

Comment:
nit: maybe we remove the prefix `_` here since these will also be shown in type completes in IDE's and it looks better without the underscore.